core/src/fields.rs

```diff
@@ -541,7 +547,7 @@ impl MongoFields {
                             if !current_col_metadata.is_empty() {
                                 self.current_col_metadata = current_col_metadata;
                                 self.current_field_for_collection = 0;
-                                return true;
+                                return Ok((true, e));
```

Comment:
by resurfacing the error below, do you mean adding the error in the other branch of this match to the (now vector) of errors?

Reply:
yes, I mean adding it to the vector of errors. I think otherwise right now it is simply dropped, no? Maybe I missed something though and it is handle somewhere and I missed it. If that's the case, ignore this ;)